sdk/trace/tracer.go

```diff
@@ -61,6 +61,13 @@ func (tr *tracer) Start(ctx context.Context, name string, o ...apitrace.SpanOpti
 	span := startSpanInternal(name, parent, remoteParent, opts)
 	span.tracer = tr
 
+	if span.IsRecordingEvents() {
```

Comment:
Is this some way of saying that the span is going to be sampled? I'm basing this question on the condition I saw for the `OnEnd` to be called - `mustExport := s.spanContext.IsSampled() && …)`. Shouldn't the condition be `s.spanContext.IsSampled()` to match the condition for calling the `OnEnd`?

Reply:
RecordingEvents and Sampling are separate. You could still be recording events even if the span is not sampled.
OnEnd() call should be fixed. It is should be called irrespective of span being sampled, IMO. It should be up to processor what to do. If the processor is used for exporting then it should check sampling flag before exporting. If the processor is store the span locally for zpages then it may ignore the sampling flag.